### PR TITLE
🐛 Fixing bmo version for clusterctl tests

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
**What this PR does / why we need it**:
Manual backport of this PR: https://github.com/metal3-io/cluster-api-provider-metal3/pull/1596 on 1.7 branch. 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partial Fix of #1832

### CODE REVIEW

**Comment by adilGhaffarDev** [comment]
@mboukhalfa please check

File: test/e2e/common.go [hunk 1]
```diff
@@ -793,3 +799,200 @@ func GetCAPM3StableReleaseOfMinor(ctx context.Context, minorRelease string) (str
...
+func BuildAndRemoveKustomization(ctx context.Context, kustomization string, clusterProxy framework.ClusterProxy) error {
+	manifest, err := buildKustomizeManifest(kustomization)
+	if err != nil {
+		return err
+	}
+	return KubectlDelete(ctx, clusterProxy.GetKubeconfigPath(), manifest)
+}
+
+// KubectlDelete shells out to kubectl delete.
+func KubectlDelete(ctx context.Context, kubeconfigPath string, resources []byte, args ...string) error {
```
Comment (mboukhalfa) [inline test/e2e/common.go:974]: Don't we have any capi framework function to delete manifests ?

File: test/e2e/common.go [hunk 2]
```diff
@@ -793,3 +799,200 @@ func GetCAPM3StableReleaseOfMinor(ctx context.Context, minorRelease string) (str
...
+	deleteCmd := testexec.NewCommand(
+		testexec.WithCommand("kubectl"),
+		testexec.WithArgs(aargs...),
+		testexec.WithStdin(rbytes),
+	)
+
+	fmt.Printf("Running kubectl %s\n", strings.Join(aargs, " "))
+	stdout, stderr, err := deleteCmd.Run(ctx)
+	fmt.Printf("stderr:\n%s\n", string(stderr))
+	fmt.Printf("stdout:\n%s\n", string(stdout))
```
Comment (mboukhalfa) [inline test/e2e/common.go:986]: we are we printing should not we send it to logs ?

File: test/e2e/config/e2e_conf.yaml [hunk 3]
```diff
@@ -166,12 +166,20 @@ variables:
...
+  BMO_RELEASE_0.6: "data/bmo-deployment/overlays/release-0.6"
+  BMO_RELEASE_LATEST: "data/bmo-deployment/overlays/release-latest"
 
 intervals:
-  default/wait-controllers: ["5m", "10s"]
+  default/wait-controllers: ["10m", "10s"]
   default/wait-cluster: ["20m", "30s"] # The second time to check the availibility of the cluster should happen late, so kcp object has time to be created
   default/wait-control-plane: ["30m", "10s"]
-  default/wait-worker-nodes: ["30m", "10s"]
+  default/wait-worker-nodes: ["60m", "10s"]
```
Comment (mboukhalfa) [inline test/e2e/config/e2e_conf.yaml:182]: Was this needed can you include it to the PRs description why we do this

**Review by mboukhalfa — APPROVED** [review]
I have some minor comments and since this already merged on main I am approving this.
Any changes you want to add can go on a follow up starting from main

File: test/e2e/common.go (hunk 1 above)
Comment (adilGhaffarDev) [reply test/e2e/common.go:974]: I was not able to find KubectlDelete in capi framework. They only have "apply" I think.

File: test/e2e/config/e2e_conf.yaml (hunk 3 above)
Comment (adilGhaffarDev) [reply test/e2e/config/e2e_conf.yaml:182]: the test was timing out here, thats why I changed it.

File: test/e2e/common.go (hunk 2 above)
Comment (adilGhaffarDev) [reply test/e2e/common.go:986]: yes I agree, I will open PR on the main with the fix and backport it. There are some other improvements too that needs to be done in this code.

**Comment by adilGhaffarDev** [comment]
@smoshiur1237 @Sunnatillo please check

File: scripts/ci-e2e.sh [hunk 4]
```diff
@@ -44,8 +44,8 @@ export IMAGE_OS=${IMAGE_OS}
 export FORCE_REPO_UPDATE="false"
 EOF
 if [[ ${GINKGO_FOCUS:-} == "features" ]]; then
-    mkdir -p "$CAPI_CONFIG_FOLDER"
-    echo "enableBMHNameBasedPreallocation: true" >"$CAPI_CONFIG_FOLDER/clusterctl.yaml"
+  mkdir -p "$CAPI_CONFIG_FOLDER"
+  echo "enableBMHNameBasedPreallocation: true" >"$CAPI_CONFIG_FOLDER/clusterctl.yaml"
```
Comment (smoshiur1237) [inline scripts/ci-e2e.sh:48]: This has been changed by the following PR: https://github.com/metal3-io/cluster-api-provider-metal3/pull/1751/files

Comment (adilGhaffarDev) [reply scripts/ci-e2e.sh:48]: That is only on the main branch, we are not backporting it.

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

File: scripts/ci-e2e.sh (hunk 4 above)
Comment (smoshiur1237) [reply scripts/ci-e2e.sh:48]: Thanks got it

**Pull request merged**